source/core/src/main/com/csse3200/game/components/player/PlayerInventoryDisplay.java

```diff
@@ -168,7 +175,12 @@ public void exit(InputEvent event, float x, float y, int pointer, Actor toActor)
             @Override
             public void changed(ChangeEvent changeEvent, Actor actor) {
                 logger.debug("Item {} was used", item.getName());
-                inventory.useItemAt(index, null);
+                ItemUsageContext context = new ItemUsageContext(entity);
+                if (item instanceof AbstractPotion) {
+                    aiComponent.addTask(new UseItemTask(entity,23, (AbstractPotion) item, context));
```

Comment:
23 is a magic number - also the naming is a bit weird, isn't this just for items that have an  effect over a period of time - probably best to change the names of abstract potion and useItemTask